### PR TITLE
US3 // part - toggle in assembler button

### PR DESCRIPTION
**Pull request by matschi3**
[US3](https://github.com/matschi3/capstone/issues/3)
[Deployment](https://capstone-git-feature-us3parttoggleinassemblerbutton-matschi3.vercel.app/)

### CODE REVIEW

File: components/PartCard/index.js [hunk 1]
```diff
@@ -4,22 +4,28 @@ import {
...
 
 export default function PartCard({ part, isDetail }) {
+  function toggleInAssembler() {
+    usePartStore.getState().togglePartValue(part.uuid, "inAssembler");
+  }
+
   return (
     <PartCardFlexContainer direction={"column"} border={"blue"}>
-      <PartCardFlexContainer direction={"row"}>
+      <PartCardFlexContainer direction={"row"} justify={"flex-start"}>
```
Comment (mpagels) [inline components/PartCard/index.js:19]: if you just use a string for a prop value you don't need the curly brackets  `{}`

File: components/PartStore/UsePartStore.js [hunk 2]
```diff
@@ -7,6 +7,18 @@ const usePartStore = create((set) => ({
...
+      parts: state.parts.map((part) =>
+        part.uuid === uuid ? { ...part, [key]: newValue } : part
+      ),
+    })),
+  togglePartValue: (uuid, key) =>
+    set((state) => ({
+      parts: state.parts.map((part) =>
+        part.uuid === uuid ? { ...part, [key]: !part[key] } : part
+      ),
+    })),
```
Comment (mpagels) [inline components/PartStore/UsePartStore.js:21]: Super awesome stuff!!

File: components/StatusMarker/index.js [hunk 3]
```diff
@@ -4,21 +4,33 @@ export default function StatusMarker({ part }) {
   return (
     <>
       {part.inAssembler === true ? (
-        <StatMarker style={{ backgroundColor: "var(--color-inAssembler)" }}>
+        <StatMarker
+          key={"inAssembler"}
+          style={{ backgroundColor: "var(--color-inAssembler)" }}
```
Comment (mpagels) [inline components/StatusMarker/index.js:9]: Don't know what `StatMarker` is. Is is a styled-components? Then you could move the style in the styled-component.

**Review by mpagels — APPROVED** [review]
Hey Nils, very very good pull request here.

Really good that you are using `zustand`.

Keep upt the good work. Have found just some small issues.

File: components/StatusMarker/index.js (hunk 3 above)
Comment (matschi3) [reply components/StatusMarker/index.js:9]: StatMarker is a reusable component, but the backgroundColor is set depending on if the status like `inAssembler` is set true

File: components/PartStore/UsePartStore.js (hunk 2 above)
Comment (matschi3) [reply components/PartStore/UsePartStore.js:21]: Yeah, i was really grinding into BearStore stuff 😁

File: components/PartCard/index.js (hunk 1 above)
Comment (matschi3) [reply components/PartCard/index.js:19]: good advice! they are gone

**Pull request merged**